snowshu/adapters/target_adapters/postgres_adapter/postgres_adapter.py

```diff
@@ -399,42 +401,31 @@ def statement_runner(statement: str):
 
     def copy_replica_data(self) -> Tuple[bool, str]:
         if self.passive_container:
-            # Copy from active to shared volume
-            logger.info('Stopping postgres in the active container...')
-            self.stop_postgres(self.container)
-            # copy the data over to a shared volume
+            # Dump from active to shared volume
             status = self.container.exec_run(
                 f"/bin/bash -c '{self.DOCKER_SHARE_REPLICA_DATA}'", tty=True)
 
-            # Copy from shared volume to passive
+            # Load dump from shared volume to passive
             self.container.stop()
             self.passive_container.start()
             logger.info('Copying replica data into passive container')
-            logger.info('Stopping postgres in passive container')
-            self.stop_postgres(self.passive_container)
 
-            logger.info("Purging passive container's pgdata dir")
-            self.passive_container.exec_run("/bin/bash -c 'rm -rf $PGDATA/*'", tty=True)
+            # Wait for db init
+            for _ in range(1200):  # for 10 minutes
+                if self.passive_container.exec_run(self.DOCKER_READY_COMMAND).exit_code == 0:
+                    break
+                time.sleep(.5)
+            else:
+                raise Exception(
```

Comment:
it is not a generic exception, I believe we should define a custom one
can be defined in the exceptions.py